src/olympia/signing/views.py

```diff
@@ -147,15 +150,26 @@ def handle_upload(self, request, addon, version_string):
                 _('Version already exists.'),
                 status.HTTP_409_CONFLICT)
 
+        package_guid = pkg.get('guid', None)
+
         dont_allow_no_guid = (
-            not addon and not pkg.get('guid', None) and
+            not addon and not package_guid and
             not pkg.get('is_webextension', False))
 
         if dont_allow_no_guid:
             raise forms.ValidationError(
                 _('Only WebExtensions are allowed to omit the GUID'),
                 status.HTTP_400_BAD_REQUEST)
 
+        if not addon and not package_guid and guid is not None:
```

Comment:
this reversed reads a lot better imho `if guid is not None and not addon and not package_guid` - don't ask me but I feel like it's making the intend of this if-statement easier to grasp.